google_compute_engine_oslogin/authorized_keys/authorized_keys.cc

```diff
@@ -37,13 +37,16 @@ int main(int argc, char* argv[]) {
   std::stringstream url;
   url << kMetadataServerUrl << "users?username=" << UrlEncode(argv[1]);
   string user_response;
-  if (!HttpGet(url.str(), &user_response)) {
+  long http_code = 0;
+  if (!HttpGet(url.str(), &user_response, &http_code)) {
     return 1;
   }
-  if (user_response.empty()) {
+  if (http_code == 404) {
     // Return 0 if the user is not an oslogin user. If we returned a failure
     // code, we would populate auth.log with useless error messages.
     return 0;
+  } else if (user_response.empty() || http_code == 500) {
```

Comment:
Good point. I don't think we even need to differentiate anything other than 404 and 200.